src/mbgl/renderer/image_manager.cpp

```diff
@@ -288,4 +288,11 @@ gfx::TextureBinding ImageManager::textureBinding(gfx::Context& context) {
     return { atlasTexture->getResource(), gfx::TextureFilterType::Linear };
 }
 
+ImageRequestor::ImageRequestor(ImageManager& imageManager_) : imageManager(imageManager_) {
+}
+
+ImageRequestor::~ImageRequestor() {
+    imageManager.removeRequestor(*this);
```

Comment:
With or without this patch, functionality seems to relly on order in renderer_impl.hpp: renderSources declared after imageManager - maybe no need for this patch.